frmts/stacit/stacitdataset.cpp

```diff
@@ -594,34 +603,35 @@ bool STACITDataset::SetupDataset(
             poVRTBand->SetColorInterpretation(eInterp);
 
         // Set band properties
-        if (asset.eoBands.IsValid() &&
-            asset.eoBands.Size() == poItemDS->GetRasterCount())
+        if (asset.bands.IsValid() &&
+            asset.bands.Size() == poItemDS->GetRasterCount())
         {
-            const auto &eoBand = asset.eoBands[i];
-            const auto osBandName = eoBand["name"].ToString();
+            const auto &band = asset.bands[i];
+            const auto osBandName = band["name"].ToString();
             if (!osBandName.empty())
                 poVRTBand->SetDescription(osBandName.c_str());
 
             if (eInterp != GCI_Undefined)
             {
-                const auto osCommonName = eoBand["common_name"].ToString();
+                auto osCommonName = band["eo:common_name"].ToString();
+                if (osBandName.empty())
```

Comment:
shouldn't that be?
```suggestion
                if (osCommonName.empty())
```

and at line 406 we check first "bands", and fallback to "eo:bands", whereas here this is first "eo:common_name", and fallback to "common_name"

Reply:
Yes, good catch.